Frontend/src/components/TableEditors/Rentals/RentalPopupFormular.svelte

```diff
@@ -73,221 +73,238 @@
-                label: "Grund",
-                group: "Pfand",
-                hidden: createNew,
-                type: InputTypes.TEXT,
-                bindTo: {
-                    keyValueStoreKey: "currentDoc",
-                    attr: "deposit_retainment_reason",
+                {
+                    id: "deposit_retainment_reason",
+                    label: "Grund",
+                    group: "Pfand",
+                    hidden: createNew,
+                    type: InputTypes.TEXT,
+                    bindTo: {
+                        keyValueStoreKey: "currentDoc",
+                        attr: "deposit_retainment_reason",
+                    },
                 },
-            },
 
-            {
-                id: "passing_out_employee",
-                label: "Ausgabe",
-                group: "Mitarbeiter",
-                type: InputTypes.TEXT,
-                bindTo: {
-                    keyValueStoreKey: "currentDoc",
-                    attr: "passing_out_employee",
+                {
+                    id: "passing_out_employee",
+                    label: "Ausgabe",
+                    group: "Mitarbeiter",
+                    type: InputTypes.TEXT,
+                    bindTo: {
+                        keyValueStoreKey: "currentDoc",
+                        attr: "passing_out_employee",
+                    },
                 },
-            },
-            {
-                id: "receiving_employee",
-                label: "Rücknahme",
-                group: "Mitarbeiter",
-                hidden: createNew,
-                type: InputTypes.TEXT,
-                bindTo: {
-                    keyValueStoreKey: "currentDoc",
-                    attr: "receiving_employee",
+                {
+                    id: "receiving_employee",
+                    label: "Rücknahme",
+                    group: "Mitarbeiter",
+                    hidden: createNew,
+                    type: InputTypes.TEXT,
+                    bindTo: {
+                        keyValueStoreKey: "currentDoc",
+                        attr: "receiving_employee",
+                    },
                 },
-            },
-            {
-                id: "remark",
-                label: "Bemerkung",
-                group: "Mitarbeiter",
-                hidden: createNew,
-                type: InputTypes.TEXT,
-                bindTo: {
-                    keyValueStoreKey: "currentDoc",
-                    attr: "remark",
+                {
+                    id: "remark",
+                    label: "Bemerkung",
+                    group: "Mitarbeiter",
+                    hidden: createNew,
+                    type: InputTypes.TEXT,
+                    bindTo: {
+                        keyValueStoreKey: "currentDoc",
+                        attr: "remark",
+                    },
                 },
-            },
-        ]);
+            ].filter((input) => !input.hidden)
```

Comment:
Your approach works well, but with this filter it would work even without the changes in all other files. I think this might be the cleaner solution.

Reply:
Ah great! good to learn, much cleaner solution. This means that the divs dont even appear within the code, right?

Reply:
Yes exactly. This array contains configuration for each input field and is passed to the formular as a prop. The formular only renders input fields that are in this configuration. This happens every time the formular opens, so it is rerendered anyway each time createNew changes.